### ISSUE TITLE
Check the results of the network

### ISSUE THREAD

**Issue by Nampv0903** [Necolizer/ISTA-Net#6]
I have trained data from some other datasets on your network, with the format of the H2O dataset, but I have not received a response when I send a request to them to check the results. So is there code to check the results?  Is there any result of this network for H2O? If so, please send it to my email: nampv0903@gmail.com, thank you

**Comment by Necolizer** [comment]
Hi @Nampv0903 , sorry I didn't get what you mean. Do you mean you trained the network on other datasets instead of H2O, but with the format of H2O? But you sent the request to H2O authors to check the results? 

1. You said 'So is there code to check the results?' Do you mean the code to check the test split results of H2O dataset? But this dataset does NOT provide the ground-truth of the test split. Therefore, it's impossible for users to write a script to get the test split results. If you want to check the test split results on H2O, you can submit your zipped result file to https://codalab.lisn.upsaclay.fr/competitions/4820#participate-submit_results . 
2. If you want to check the results on some other datasets, just follow the instructions provided by their README file.
3. You said 'Is there any result of this network for H2O?' Do you mean the results in our own experiments? We have reported it in our paper.

**Comment by Nampv0903** [comment]
Thank you for your reply, I initially tested this network on H2O database
and submitted the results for testing.  While waiting for a response, I
tested this network on other data sets in the format of the H2O data set,
but so far my request has not received a response, so I would like a script
to test it.  Check the results of the network.  As you said, I can submit
my zipped result file to
https://codalab.lisn.upsaclay.fr/competitions/4820#participate-submit_results.
So can I send the results of the network I trained with different data sets
to check the results?

Vào Th 5, 29 thg 8, 2024 vào lúc 18:23 Necolizer ***@***.***>
đã viết:

**Comment by Necolizer** [comment]
I see. In this case, you can NOT send the results of the network which you trained with different datasets
to [that Codalab Link](https://codalab.lisn.upsaclay.fr/competitions/4820#participate-submit_results), because this link is for **H2O only**.

Sorry, I don't have such script to evaluate other datasets, because there's no such need in my experiments. Here's some suggestions:
- I think the easiest way to get the top-1 accuracy of test split of your 'other datasets' is to modify [the train config yaml file](https://github.com/Necolizer/ISTA-Net/blob/main/config/h2o/h2o.yaml). You can replace the validation set configurations (e.g., data_path) with your test set ones, if you have the ground-truths (labels) of the test sets in your 'other datasets'. Run the training process once again, then you will see the test scores.
- Another way is to modify [the test config yaml file](https://github.com/Necolizer/ISTA-Net/blob/main/config/h2o/h2o_get_test_results.yaml) and [Feeder_Test](https://github.com/Necolizer/ISTA-Net/blob/main/feeders/feeder_h2o.py). Replace 'run_mode: h2o_test_get_results' with 'run_mode: test'. Then you should provide your ground-truth in Feeder_Test class. In this way, you run the testing process to see the test scores.

Hope this will help.

Best,

**Comment by Nampv0903** [comment]
I understand, thank you for your support!
